e2e/upgrade_test.go

```diff
@@ -0,0 +1,97 @@
+package e2e
+
+import (
+	"flag"
+	"os"
+	"testing"
+	"time"
+
+	tools "github.com/elastic/elastic-agent/e2e/tools"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gexec"
+)
+
+var clusterConfigPath string
+var clusterConfig tools.ClusterConfig
+var client *tools.Client
+var agentVersion = os.Getenv("AGENT_VERSION")
+
+func init() {
+	flag.StringVar(&clusterConfigPath, "config", "./cluster-digest.yml", "Path to cluster config")
+}
+
+func TestElasticAgentUpgrade(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Basic Suite")
+}
+
+var _ = BeforeSuite(func() {
+	By("validating flags and config")
+	Expect(clusterConfigPath).NotTo(BeZero(), "Make sure --config is set correctly.")
+
+	c, err := tools.ReadConfig(clusterConfigPath)
+	Expect(err).NotTo(HaveOccurred())
+	clusterConfig = c
+	client, err = tools.NewClient(&clusterConfig)
+	Expect(err).NotTo(HaveOccurred())
+})
+
+var _ = Describe("Smoketests", func() {
+
+	Describe("Elastic Agent Install and Upgrade", Ordered, func() {
```

Comment:
Can we remove the BDD framework here and just write plain go tests? We expect only engineers will be writing tests and BDD gets in the way more than it helps.

If you need a suite abstraction to coordinate setup and teardown https://pkg.go.dev/github.com/stretchr/testify/suite is a good option.

Reply:
+1 to this. I really don't think BDD is a win for E2E testing and I think it hindered adoption of the original framework.

Reply:
Let me give some additional context. It's not BDD (Given-When-Then notation). [Ginkgo](https://github.com/onsi/ginkgo) supports various notations including BDD but we don't use it here. 
It this case we use simple notation (IIRC it initially comes from [mocha.js](https://mochajs.org)).

`Describe()` - represents a test suite name
`It()` - declared a test case

For example : 
```
Describe("Elastic Agent Install and Upgrade", func() {
    It("is online after upgrade", func() {
        ...
    })
)}
```
`Describe` blocks can be folded into each other, however we can keep them plain, i.e. use a single `Describe` for test suites.

[Kubernetes e2e tests](https://github.com/kubernetes/kubernetes/tree/master/test/e2e) use exactly this approach: Ginkgo + one level `Describe` blocks. 

Probably the default go test is better. It's up to you. Just want to make sure that we are on the same page. 
To do that I need to substitute [Gomega matchers](https://onsi.github.io/gomega/) to Testify.

Reply:
Let's start with just the basic Go tests, I believe that is the preference of most of our engineers.

I want to focus on getting the e2e test framework into a reliably usable state and we can evaluate new testing frameworks and techniques later. There is probably value in Gingko but I don't want it to get in the way of the initial tests.

Reply:
Ok, on it

Reply:
Done